Adding an upgrades and version support section to troubleshooting (#765)
minio/docs: source/operations/troubleshooting.rst

Upgrades and Version Support
----------------------------

MinIO regularly releases updates to introduce features, improve performance, address security concerns, or fix bugs. 
These releases can occur very frequently, and vary by product.

Always test software releases in a development environment before upgrading on a production deployment.

Active Support Periods
~~~~~~~~~~~~~~~~~~~~~~

Version support varies by the `license <https://min.io/pricing?ref=docs>`_ used for the deployment.

.. list-table:: 
   :header-rows: 1
   :widths: 30 70
   :width: 100%  

   * - License
     - Support length 

   * - AGPLv3
     - Most recent release

   * - MinIO Standard
     - 1 year long term support of any release

   * - MinIO Enterprise
     - 5 year long term support of any release, SUBNET support for upgrade guidance and recommendations

Recommended Upgrade Schedule
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

MinIO recommends always installing the most recent release to obtain the latest features, security enhancements, and performance improvements.
We recognize that such a frequent release schedule may make this impractical for some organizations.

> In such cases, we recommend using MinIO and other product releases that are no older than six months.

Reviewer: otherwise this reads like guidance for all other products.

> In such cases, we recommend using MinIO and other product releases that are no older than six months.

Author: Going to keep it here and use your wording suggestion.

Use :mc:`mc admin logs` command to display logs from the command line.
The command supports type and quantity filters for further limiting logs output.

Upgrades and Version Support
----------------------------

> MinIO releases versions of its software as often as needed to introduce features, improve performance, address security concerns, or fix bugs. 

Reviewer: ```suggestion
MinIO regularly releases  updates to introduce features, improve performance, address security concerns, or fix bugs. 
```

These releases can occur very frequently, and vary by product.

> Always test upgrades in a development environment before upgrading on a production deployment.

Reviewer: Q: Do we want to use "Upgrades" or "Updates" ? 

I have always thought of "Upgrade" as the verb and "Update" as the thing, but I guess they are interchangeabl?

Reviewer: But here, "Test upgrades before upgrading" -> "Test updates before upgrading" or "Test new software releases...before upgrading"

Author: Going with "Always test software releases ..."

> Version Support

Reviewer: Active Support Periods <- I think this is a more recognizable term

~~~~~~~~~~~~~~~

Version support varies by the `license <https://min.io/pricing?ref=docs>`_ used for the deployment.

.. list-table:: 
   :header-rows: 1
   :widths: 30 70
   :width: 100%  

   * - License
     - Support length 

   * - AGPLv3
     - Most recent release

   * - MinIO Standard
     - 1 year long term support of any release

   * - MinIO Enterprise
     - 5 year long term support of any release, SUBNET support for upgrade guidance and recommendations

> Version Updates

Reviewer: Maybe "Recommended Update Schedule" ?

~~~~~~~~~~~~~~~

MinIO recommends always installing the most recent release to obtain the latest features, security enhancements, and performance improvements.
We recognize that such a frequent release schedule may make this impractical for some organizations.
In such cases, we recommend using MinIO and other product releases that are no older than six months.

> From time to time, we may backport some trivial changes to older releases to best support licensed customers.

Reviewer: They are also sometimes not that trivial at-all, and those backports are not always publicly available.

I would recommend dropping this and the next graf entirely.

> From time to time, we may backport some trivial changes to older releases to best support licensed customers.

Author: Dropped.
